x/migration/fs_source_test.go

```diff
@@ -0,0 +1,46 @@
+// +build go1.16
+
+package migration
+
+import (
+	"context"
+	"testing"
+	"testing/fstest"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFSSource(t *testing.T) {
+	var (
+		ctx = context.Background()
+		fs  = fstest.MapFS{
+			"3_final.down.sql":      &fstest.MapFile{Data: []byte("bar")},
+			"2_initial.up.postgres": &fstest.MapFile{Data: []byte("foo")},
+			"3_final.up.sql":        &fstest.MapFile{Data: []byte("bar")},
+		}
+	)
+
+	s, err := NewFSSource(fs, ".")
+	require.NoError(t, err)
```

Comment:
Can you make a test with  an error now that you are returning one :smile:

Reply:
Fair.